docs/apache-airflow/issue-triage-process.rst

```diff
@@ -0,0 +1,195 @@
+“Airflow Core” area of the project.
+
+Similarly, some providers are broken out such as Apache (``provider/Apache``),
+AWS (``provider/AWS``), Azure (``provider/Azure``), and Google (``provider/Google``).
+These make it easier for developers working on a single provider to
+track issues for that provider.
+
+Most issues need a combination of "kind" and "area" labels to be actionable.
+For example:
+
+* Feature request for an additional API would have ``kind:feature`` and ``area:API``
+* Bug report on the User Interface would have ``kind:bug`` and ``area:UI``
+* Documentation request on the Kubernetes Executor, would have ``kind:documentation`` and ``area:kubernetes``
+
+
+Response to issues
+''''''''''''''''''
+
+Once an issue has been created on the Airflow project, someone from the
+Airflow team or the Airflow community typically responds to this issue.
+This response can have multiple elements.
+
+**Priority**
+
+After significant discussion about the different priority schemes currently
+being used across various projects, we decided to use a priority scheme based
+on the Kubernetes project, since the team felt it was easier for people to
+understand.
+
+Therefore, the priority labels used are:
+
+* ``priority:critical``: Showstopper bug that should be resolved immediately and a patch issued as soon as possible. Typically, this is because it affects most users and would take down production systems.
+* ``priority:high``: A high priority bug that affects many users and should be resolved quickly, but can wait for the next scheduled patch release.
+* ``priority:medium``: A bug that should be fixed before the next release, but would not block a release if found during the release process.
+* ``priority:low``: A bug with a simple workaround or a nuisance that does not stop mainstream functionality.
+
+
+It's important to use priority labels effectively so we can triage incoming issues
+appropriately and make sure that when we release a new version of Airflow,
+we can ship a release confident that there are no “production blocker” issues in it.
+
+
+**Milestones**
+
+The key temporal element in the issue triage process is the concept of milestones.
+This is critical for release management purposes and will be used represent upcoming
+release targets.
+
+Issues currently being resolved will get assigned to one of the upcoming releases.
+For example a feature request may be targeted for the next feature release milestone
+such as ``2.x``, where a bug may be targeted for the next patch release milestone
+such as ``2.x.y``.
+
+In the interest of being precise, when an issue is tagged with a milestone, it
+represents that it will be considered for that release, not that it is committed to
+a release. Once a PR is created to fix that issue and when that PR is tagged with a
+milestone, it implies that the PR is intended to released in that milestone.
+
+It is worth noting that Airflow Core and Airflow Providers are now released and
```

Comment:
I am not sure if we want to actually have milestones for Providers. We could have a separate milestone for every batch of providers for sure, but I think this is not necessary, as we will always release whatever we have in master as the next wave of providers. What is more important is a list of upcoming changes in providers - so an automated way to show what will be included in the next wave of providers (so everything that is merged already) - and we will have a separate issue which will get updated automatically with:

* list of providers that we will release (i.e. all providers that have some changes)
* for each provider, detailed list of changes that we will release

This list will be automatically updated and once we release a next wave of providers, it will clean-up automatically.

I believe this is a bit different than core Airflow. For Airflow we deliberately choose what we are going to release by cherry-picking (at least for patchlevels), where for Providers we always release everything we have in master.

Reply:
Hmm, that's a good point. 
I was basing this on the behavior to date including the milestone "apache-airflow-providers-google v2" which is currently there in github issues. 
I think you are making the point, that going forward this should be an exception rather than the rule. 
Did I understand you correctly?

Reply:
Correct. I am planning to propose how formally our provider release process will look like (basically release all provider that changed from master every 3 weeks in regular intervals (RC - 1 week vote  - Release - 2 weeks break) - i wanted to discuss it at the dev call today. and in this case we can set a release schedule in calendar rather than a milestone.

Reply:
Done. I explicitly made the milestone reference only towards Airflow Core releases now. Can update this later once the Provider release proposal is locked down.